### PR TITLE
@alloy => V2 artist app

### PR DESCRIPTION
**Pull request by mzikherman**
Trying to get the artist app to load in storybooks.

Currently working on the pagination/size arguments to the new connection style for the filter.

### CODE REVIEW

File: src/Apps/Artist/routes.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { ShowsRouteFragmentContainer as ShowsRoute } from "./Routes/Shows"
 // * `Redirect` needs to be casted, as it’s not compatible with `RouteConfig`
 export const routes: RouteConfig[] = [
   {
-    path: "/artist/:artist_id",
+    path: "/artist/:artistID",
```
Comment (damassi) [inline src/Apps/Artist/routes.tsx]: @mzikherman - When i was building the new filter I discovered that this has to be `artist_id` as the form of the variable is passed along to the query made to gravity / elastic search.

File: src/Apps/Artist/routes.tsx [hunk 2]
```diff
@@ -101,7 +101,7 @@ export const routes: RouteConfig[] = [
       // and not get caught here.
       new Redirect({
         from: "*",
-        to: "/artist/:artist_id",
+        to: "/artist/:artistID",
```
Comment (damassi) [inline src/Apps/Artist/routes.tsx]: Same here. Not sure if something changed on the MP side to account for this, though.

File: src/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx [hunk 3]
```diff
@@ -65,7 +71,7 @@ const ArtistArtworkFilter: React.FC<ArtistArtworkFilterProps> = props => {
 }
 
 export const ArtistArtworkFilterRefetchContainer = createRefetchContainer(
-  ArtistArtworkFilter,
+  withRouter(ArtistArtworkFilter),
```
Comment (mzikherman) [inline src/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx]: This was a breaking change in 0.4 of `found`. Without this HOC, the `match` prop didn't seem to exist (and `location` also wasn't present as it was no longer spread out of `match` - https://github.com/relay-tools/found-relay/pull/163).

I still don't get why we now need `withRouter`, but that seems to be the documented way in 0.4.

File: src/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx [hunk 4]
```diff
@@ -120,7 +126,8 @@ export const ArtistArtworkFilterRefetchContainer = createRefetchContainer(
           page: $page
           partnerID: $partnerID
           priceRange: $priceRange
-          size: 0
+          first: 30
```
Comment (mzikherman) [inline src/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx]: hard code for now

File: src/Apps/Artist/routes.tsx (hunk 1 above)
Comment (mzikherman) [reply src/Apps/Artist/routes.tsx]: Yea, I think the code mods in V2 may have made that all work, this seemed to be fine. Everything, included the filter , is nested under `artist`. And the filter should automatically get the `artist_id` param added from the parent by the resolver.

Comment (damassi) [reply src/Apps/Artist/routes.tsx]: Cool, just want to confirm. Before it was `artistID`, but with the new dynamicism of the rebulit artwork filter some of the values being passed to the backend are implicit, and `found` will take the variable form from the route defintion (`:artist_id`) and pass it directly over. Worth double checking that everything works as expected and if so 👍

**Comment by mzikherman** [comment]
Merging for now! (filter pagination not working).

File: src/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx (hunk 3 above)
Comment (alloy) [reply src/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx]: `withRouter` plucks the `match` value out of a React context, provided by the router, and then injects it as a prop into the wrapped component

**Comment by artsyit** [comment]
:rocket: PR was released in v22.0.0 :rocket:

**Pull request merged**